### ISSUE TITLE
6 errors in tests

### ISSUE THREAD

**Issue by yurivict** [epics-base/setuptools_dso#32]
```
========================================================================================= FAILURES ==========================================================================================
_______________________________________________________________________________________ test session ________________________________________________________________________________________
[gw3] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11
mypy exited with status 1.
______________________________________________________________________________ src/setuptools_dso/__init__.py _______________________________________________________________________________
[gw0] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11
65: error: Skipping analyzing "Cython.Build": module is installed, but missing library stubs or py.typed marker  [import-untyped]
65: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
________________________________________________________________________________ src/setuptools_dso/probe.py ________________________________________________________________________________
[gw2] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11
17: error: Name "TemporaryDirectory" already defined (possibly by an import)  [no-redef]
_______________________________________________________________________________ src/setuptools_dso/dsocmd.py ________________________________________________________________________________
[gw3] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11
14: error: Skipping analyzing "wheel.bdist_wheel": module is installed, but missing library stubs or py.typed marker  [import-untyped]
27: error: Incompatible import of "_build" (imported name has type "type[distutils.command.build.build]", local name has type "type[setuptools.command.build.build]")  [assignment]
43: error: "type[Distribution]" has no attribute "x_dsos"  [attr-defined]
52: error: Name "Pool" already defined (possibly by an import)  [no-redef]
___________________________________________________________________________________ TryCompile.test_info ____________________________________________________________________________________
[gw0] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <setuptools_dso.test.test_probe.TryCompile testMethod=test_info>

    def test_info(self):
        info = self.probe.info
        print("Raw Macros", info._raw_macros)
        print("Info", info)
    
        self.assertIn(info.compiler, ('clang', 'gcc', 'msvc'))
        self.assertGreater(info.compiler_version, (0,))
>       self.assertIn(info.target_os, ("cygwin", "linux", "osx", "windows"))
E       AssertionError: None not found in ('cygwin', 'linux', 'osx', 'windows')

src/setuptools_dso/test/test_probe.py:73: AssertionError
----------------------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------------------
cpp -o /tmp/tmpfz5sn2rj/eval_macros_out.c /tmp/tmpfz5sn2rj/eval_macros_in.c
/* test compile */
#define PROBESIZE (sizeof(void*))
char probe_info[] = {'P','R','O','B','E','I','N','F','O','[',  ('0'+((PROBESIZE/10000)%10)),  ('0'+((PROBESIZE/1000)%10)),  ('0'+((PROBESIZE/100)%10)),  ('0'+((PROBESIZE/10)%10)),  ('0'+((PROBESIZE/1)%10)),']'};

creating /tmp/tmpfz5sn2rj/tmp
creating /tmp/tmpfz5sn2rj/tmp/tmpfz5sn2rj
cc -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -fPIC -c /tmp/tmpfz5sn2rj/try_compile.c -o /tmp/tmpfz5sn2rj/tmp/tmpfz5sn2rj/try_compile.o
Probe sizeof(void*) = 8
Raw Macros OrderedDict([('__clang__', '1'), ('__clang_major__', '18'), ('__clang_minor__', '1'), ('__clang_patchlevel__', '5'), ('__GNUC__', '4'), ('__GNUC_MINOR__', '2'), ('__GNUC_PATCHLEVEL__', '1'), ('_MSC_VER', None), ('_MSC_FULL_VER', None), ('__APPLE__', None), ('__CYGWIN__', None), ('__linux__', None), ('_WIN32', None), ('__aarch64__', None), ('__arm__', None), ('__x86_64__', '1'), ('__i386__', None), ('_M_ARM', None), ('_M_ARM64', None), ('_M_AMD64', None), ('_M_X64', None), ('_M_IX86', None), ('__ARMEB__', None), ('__ARMEL__', None), ('__AARCH64EB__', None), ('__AARCH64EL__', None)])
Info ToolchainInfo(address_width=64, compiler='clang', compiler_type='unix', compiler_version=(18, 1, 5), endian='little', gnuish=True, target_arch='amd64', target_os=None)
----------------------------------------------------------------------------------- Captured stderr call ------------------------------------------------------------------------------------
Warning: unable to classify target_os

-------- coverage: platform freebsd14, python 3.11.9-final-0 ---------
Name                                       Stmts   Miss  Cover
--------------------------------------------------------------
src/setuptools_dso/__init__.py                56     44    21%
src/setuptools_dso/compiler.py                48     30    38%
src/setuptools_dso/dsocmd.py                 364    282    23%
src/setuptools_dso/probe.py                  187     37    80%
src/setuptools_dso/runtime.py                 79     59    25%
src/setuptools_dso/test/__init__.py            0      0   100%
src/setuptools_dso/test/test_parallel.py       6      0   100%
src/setuptools_dso/test/test_probe.py         41      3    93%
--------------------------------------------------------------
TOTAL                                        781    455    42%

=========================================================================================== mypy ============================================================================================
Found 6 errors in 3 files (checked 9 source files)
=============================================================================== 5 failed, 14 passed in 23.67s ===============================================================================
*** Error code 1
```

Version: 2.10
Python-3.11
FreeBSD 14.1

**Comment by mdavidsaver** [comment]
> >       self.assertIn(info.target_os, ("cygwin", "linux", "osx", "windows"))
> E       AssertionError: None not found in ('cygwin', 'linux', 'osx', 'windows')

@yurivict Thank you for reporting.  Failure on FreeBSD isn't surprising.  As the above code indicates by omission, I did not consider *BSD when creating setuptools_dso.  (I run Linux exclusively on my systems)  Some of these errors may be fixed by #31, however I some additional development would be required to achieve full support.

If someone is interested in attempting to add FreeBSD support.  Well.  PRs welcomed.

While I would be delighted to see GHA jobs for this OS, I know this isn't trivial.  Or at least wasn't the last time I looked.  So I can not in good consensus make this a requirement.
